dockerspawner/dockerspawner.py

```diff
@@ -185,34 +187,42 @@ def poll(self):
         container = yield self.get_container()
         if not container:
             self.log.warn("container not found")
-            raise gen.Return(0)
-        status = container['Status']
-        self.log.debug("Container %s status: %s", self.container_id[:7], status)
-        parts = status.split()
-        # status examples:
-        # 'Exited (127) 12 days ago'
-        # 'Up 4 days'
-        # 'Up 4 days (Paused)' <- TODO: this isn't handled
-        # e.g.
-        if not parts or parts[0] not in ('Up', 'Exited'):
-            raise ValueError("Unhandled status: '%s'" % status)
-        if parts[0] == 'Up':
-            raise gen.Return(None)
-        elif parts[0] == 'Exited':
-            raise gen.Return(int(parts[1][1:-1]))
+            return ""
+
+        container_state = container['State']
+        self.log.debug(
+            "Container %s status: %s",
+            self.container_id[:7],
+            pformat(container_state),
+        )
+
+        if container_state["Running"]:
+            return None
+        else:
+            return (
+                "ExitCode={ExitCode}, "
+                "Error='{Error}', "
+                "FinishedAt={FinishedAt}".format(**container_state)
```

Comment:
Is there any chance of KeyErrors in this format call? A previous bug could cause weird state when docker was having a hard time. Should we be less trusting of docker here? (AOK if the answer is no)